featuretools/entityset/entityset.py

```diff
@@ -715,6 +715,13 @@ def normalize_entity(self, base_entity_id, new_entity_id, index,
             if v == index:
                 raise ValueError("Not copying {} as both index and variable".format(v))
                 break
```

Comment:
We should put at the top of this file the following, so we correctly check that the type is a string in python 2 and 3.
```python
if is_python_2():
    string_types = basestring
else:
    string_types = str
```
Then below, we can check the following
```python
if isinstance(make_time_index, string_types):
```
This will help us deal with unicode issues with python 2 and python 3

Reply:
checking the `string_types = basestring` comes up with an error when I make lint. Should I put some kind of lint ignore statement there? Or define basestring at some point earlier?

Reply:
Actually, we can do the following
```python
from featuretools.utils import is_string

if is_string(make_time_index):
```